uint32 port
uint32 pin
bool state
---
bool success